samples/cpp/gstreamer_pipeline.cpp

```diff
@@ -0,0 +1,207 @@
+    {
+        cout << "Command processor doesn't exists" << endl;
+    }
+
+    // Choose a video container
+    if (arg_container == "avi")
+    {
+        container_plugin = "avidemux";
+    }
+    else if (arg_container == "mp4"  || arg_container == "mov")
+    {
+        container_plugin = "qtdemux";
+    }
+    else if (arg_container == "mkv")
+    {
+        container_plugin = "matroskademux";
+    }
+    else
+    {
+        help();
+        cout << "Not find container type" << endl;
+        return -2;
+    }
+
+    // Choose the constructed GStreamer pipeline
+    if (arg_plugin == "gst-default")
+    {
+        string pipeline = "filesrc location=\"" + arg_file_name + "\" \
+        ! " + container_plugin + "                    \
+        ! decodebin                                   \
+        ! videoconvert                                \
+        ! appsink sync=false";
+        cap = VideoCapture(pipeline, CAP_GSTREAMER);
+    }
+    else if (arg_plugin == "gst-vaapi")
+    {
+        string pipeline = "filesrc location=\"" + arg_file_name + "\" \
+        ! " + container_plugin + "                    \
+        ! vaapidecodebin  disable-vpp=true            \
+        ! deinterlace                                 \
+        ! autovideoconvert                            \
+        ! videoconvert                                \
+            dither=0                                  \
+            dither-quantization=0                     \
+            chroma-resampler=0                        \
+            alpha-mode=1                              \
+            alpha-value=0                             \
+            chroma-mode=3                             \
+        ! appsink sync=false";
+        cap = VideoCapture(pipeline, CAP_GSTREAMER);
+    }
+    else if (arg_plugin == "gst-libav")
+    {
+        // Choose a codec for libav
+        if (arg_container == "mpeg4")
+        {
+            codec_name = "avdec_mpeg4";
+        }
+        else if (arg_container == "h264")
+        {
+            container_plugin = "avdec_h264";
+        }
+        else if (arg_container == "h265")
+        {
+            container_plugin = "avdec_h265";
+        }
+        else
+        {
+            help();
+            cout << "Not find codec" << endl;
+            return -3;
+        }
+
+        string pipeline = "filesrc location=\"" + arg_file_name + "\" \
+        ! " + container_plugin + "                    \
+        ! " + codec_name + "                          \
+        ! videoconvert                                \
+        ! appsink sync=false";
+        cap = VideoCapture(pipeline, CAP_GSTREAMER);
+    }
+    else if (arg_plugin == "ffmpeg")
+    {
+        cap = VideoCapture(arg_file_name, CAP_FFMPEG);
+    }
+    else
+    {
+        help();
+        cout << "Not find GStreamer pipeline" << endl;
+        return -4;
+    }
+
+    // Choose a show video or mesuare fps
+    if (arg_start == "show_video")
```

Comment:
I think we can join these two modes into one with a flag:
```.cpp
bool showFrames = <read from cmdline>;
//...
int64 tick = getTickCount();
cap >> frame;
tick = getTickCount() - tick;
if (showFrames)
    imshow("Frame", frame); // it is not necessary to call namedWindow(), it will be done automatically
//...
```